ballerina/msg_listener.bal

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.org).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.runtime;
+import ballerina/log;
+import ballerina/task;
+
+# Represents the default error type.
+public type Error distinct error;
+
+# Represents the message store listener configuration,
+public type StoreListenerConfiguration record {|
+    # The interval in seconds at which the listener polls for new messages
+    decimal pollingInterval = 1;
+    # The maximum number of retries for processing a message. 
+    # If set to 0, the message will not be retried
+    int maxRetries = 3;
+    # The interval in seconds between retries for processing a message
+    decimal retryInterval = 1;
+    # If true, the message will be dropped after the maximum number of retries is reached
+    boolean dropMessageAfterMaxRetries = false;
+    # An optional message store to store messages that could not be processed after the maximum 
+    # number of retries. When set, `dropMessageAfterMaxRetries` will be ignored
+    Store deadLetterStore?;
+|};
+
+# Represents a message store listener that polls messages from a message store and processes them.
+public isolated class StoreListener {
```

Comment:
Any reason to name this as `StoreListener` rather than using `Listener` ?

Reply:
We wanted the name to be more specific to indicate its a listener which is related to a store. Since the package is more generic(`messaging`), this makes more sense than just `Listener`

Reply:
Discussed the reason during the offline code review